lib/charms/prometheus_k8s/v0/prometheus_remote_write.py

```diff
@@ -681,9 +684,37 @@ def _push_alerts_to_relation_databag(self, relation: Relation) -> None:
 
         alert_rules_as_dict = alert_rules.as_dict()
 
+        alert_rules_as_dict = self._parse_and_add_extra_alerts(alert_rules_as_dict)
+
         if alert_rules_as_dict:
             relation.data[self._charm.app]["alert_rules"] = json.dumps(alert_rules_as_dict)
 
+    def _parse_and_add_extra_alerts(self, alert_rules_as_dict: dict) -> dict:
+        """Parse alerts from `self.extra_alerts_callable`.
+
+        Call `self.extra_alerts_callable`, parse alert groups and
+        extend (mutate) `alert_rules_as_dict` with new alerts.
+
+        Returns:
+            (dict) Modified `alert_rules_as_dict`. Return is done only for explicit mutation.
+        """
+        if callable(self.extra_alerts_callable):
+            extra_alerts_list = []
+            extra_alerts = self.extra_alerts_callable()
+            if extra_alerts:
+                for topology_identifier, alert_rule_groups in extra_alerts.items():
+                    extra_alerts_list.extend(alert_rule_groups.get("groups", []))
+
+                if alert_rules_as_dict:
+                    alert_rules_as_dict["groups"].extend(extra_alerts_list)
+                else:
+                    alert_rules_as_dict["groups"] = extra_alerts_list
+
+                logger.debug(
+                    "Extra rules were pushed to remote write endpoint. %s", extra_alerts_list
```

Comment:
Not sure if we want to dump all this, even if it's just debug. Perhaps:
```suggestion
                    "%s extra rules were pushed to remote write endpoint", len(extra_alerts)
```
?

Reply:
:+1:

Reply:
why `len(extra_alerts)` ?
this is a dict and number of keys will represent number of groups and not number of rules, or?

Reply:
probably?

```python
logger.debug(
                    "%s extra rules were pushed to remote write endpoint. %s", len(extra_alerts_list)
                )
```

Reply:
fixed